src/test/java/edu/usf/cutr/pelias/SearchTest.java

```diff
@@ -193,5 +193,21 @@ public void testRequestParameters() throws IOException {
                         "&boundary.rect.min_lat=27.959868&boundary.rect.min_lon=-82.515286" +
                         "&boundary.rect.max_lat=28.131471&boundary.rect.max_lon=-82.367646",
                 request.getUrl().toString());
+
+        request = new SearchRequest.Builder(API_KEY, "burger king")
+                .setApiEndpoint(SEARCH_WITH_FOCUS_ENDPOINT)
+                .setFocusPoint(28.061062d, -82.4132d)
+                .setSources("osm")
+                .setSize(35)
+                .setBoundaryRect(27.959868, -82.515286, 28.131471, -82.367646)
+                .build();
+
+        assertEquals("https://raw.githubusercontent.com/CUTR-at-USF/pelias-client-library/master/src/test/resources/" +
+                        "search-with-focus.json?text=burger+king&api_key=dummyApiKey&sources=osm&size=35" +
```

Comment:
Hmm I don't think the guava UrlEscapers will work as well as UrlEncoder because it seems like UrlEncoder treats more characters as unsafe - particularly "+" and "&" which are escaped as %xy encodings to prevent injecting arbitrary parameters. So, with Guava it would pass "subway&size=10000" through to the URL.  This applies to both the urlPathSegmentEscaper and urlFragmentEscaper.  

The reason why those two are less strict is because they seem to be designed for escaping http://mobullity.forest.usf.edu/path/to#fragment  where the path is each segment (between '/') of the "/path/to" part, and the fragment is everything following the "#" which can be nearly anything.


urlFormParameterEscaper looks like it works the same as UrlEncoder, but it may be allow CR+LF sequences since it is used for escaping form parameters (HTTP POST vs. HTTP GET)